### ISSUE TITLE
`no-implicit-this` + strict mode false positives

### ISSUE THREAD

**Issue by Windvis** [ember-template-lint/ember-template-lint#2785]
I experimented with ember-template-imports again now that the ["Road to stable" issue](https://github.com/ember-template-imports/ember-template-imports/issues/35) has been marked as completed, but I'm running into a lot of no-implicit-this false positives. 

According to [some comments on the original PR](https://github.com/ember-template-lint/ember-template-lint/pull/2483#discussion_r858957036), it seems the intended behaviour is to disable this rule if it's a strict mode template.

However, [this check](https://github.com/ember-template-lint/ember-template-lint/blob/0b4983a496b40452d3747a921c062cc5fa50c336/lib/rules/no-implicit-this.js#L56) seems to be inversed? On quick glance it also seems that the `this.isStrictMode` value [isn't always correctly set either](https://github.com/ember-template-lint/ember-template-lint/blob/0b4983a496b40452d3747a921c062cc5fa50c336/lib/extract-templates.js#L96-L97). Normal .hbs files also seem to have `this.isStrictMode` set to true.

I added a failing test in [this PR](https://github.com/ember-template-lint/ember-template-lint/pull/2784).

**Comment by Techn1x** [comment]
I have also hit this issue today.

With a glint file like below, `Foo` is flagged as an ambiguous reference

```gts
import type { TOC } from '@ember/component/template-only'
import { Foo } from 'my-app/foos/foo'
import SomeComponent from 'my-app/components/some-component'

export interface Signature {}

export const MyComponent: TOC<Signature> = <template>
<div>
  <SomeComponent @foo={{Foo}} />
</div>
```

**Comment by NullVoxPopuli** [comment]
As a temporary work around, disabling the lint is an option.


I think i know how to make it work for gjs tho

**Comment by NullVoxPopuli** [comment]
I had original thought that the build of gjs/gts files would protect us from needing this lint, but gjs/gts files are still compiled fine with missing references, deferring the error to boot-time (when modules start loading).

So this lint _does_ need fixing.

**Comment by NullVoxPopuli** [comment]
also, the `no-implicit-this` lint rule was already disabled in this PR: https://github.com/ember-template-lint/ember-template-lint/pull/2844
released in: 5.7.1: https://github.com/ember-template-lint/ember-template-lint/blob/master/CHANGELOG.md#v571-2023-03-16

So we can close this issue, @bmish

**Comment by Techn1x** [comment]
> As a temporary work around, disabling the lint is an option.

Disabling the no-implicit-this rule for gjs/gts files is exactly what I'm doing as a workaround (and it works). But from what I can tell, I shouldn't have to disable that rule myself.

I don't think this issue should be closed - I experienced it just yesterday, and I am on the latest ember-template-lint 5.7.3 - am I correct in saying that my usage of `Foo` in my code snippet from before should not cause a violation of `no-implicit-this` ?

If we don't fix this, it's a point of confusion and may make users create unnecessary javascript-backed templates when a template-only component is meant to suffice.

**Comment by NullVoxPopuli** [comment]
I found a reproduction!
```js
'use strict';

module.exports = {
  plugins: [],
  extends: ['recommended'],
};
```
this is the default config, and the overrides do not apply.

**Comment by NullVoxPopuli** [comment]
ok, so my entire "not able to reproduce silliness" was just because all my template-lintrcs were too non-standard.

Looking at the `processExtends` method in ember-template-lint, _we don't even include the `overrides`!!_

_This is quite the bug!_

**Comment by NullVoxPopuli** [comment]
Got a fix here: https://github.com/ember-template-lint/ember-template-lint/pull/2891

**Comment by Techn1x** [comment]
Thanks a bunch for looking into it :1st_place_medal:

**Comment by NullVoxPopuli** [comment]
thanks for debuggin with me <3

**Comment by Techn1x** [comment]
confirmed issue resolved in 5.8.0, many thanks

**Comment by ynotdraw** [comment]
Thanks so much for resolving this @NullVoxPopuli!
